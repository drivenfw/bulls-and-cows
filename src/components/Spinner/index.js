import styled, { css, keyframes } from 'styled-components'


const rotate360 = keyframes`
  from { transform: rotate(0deg) }
  to { transform: rotate(360deg) }
`

export default styled.div`
  width: 150px;
  height: 150px;
  border-radius: 50%;

  ${({ theme }) => css`
    border: 5px solid ${theme.primaryColor2};
    border-top-color: ${theme.primaryColor};
  `}

  animation: ${rotate360} 1s linear infinite;
`

