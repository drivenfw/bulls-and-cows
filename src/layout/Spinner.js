import styled, { css, keyframes } from 'styled-components'


const rotate360 = keyframes`
  from { transform: rotate(0deg) }
  to { transform: rotate(360deg) }
`

export default styled.div`
  width: 150px;
  height: 150px;
  border-radius: 50%;

  ${({ theme }) => css`
    border: 5px solid ${theme.primaryColor2};
    border-top-color: ${theme.primaryColor};
  `}

  animation: ${rotate360} 1s linear infinite;

  @media (min-width: 850px) {
    width: 250px;
    height: 250px;
    border-width: 10px;
  }
`

