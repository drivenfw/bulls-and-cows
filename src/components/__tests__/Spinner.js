import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'

import Spinner from 'components/Spinner'


configure({ adapter: new Adapter() })

describe('<Spinner />', () => {
  describe('snapshot test', () => {
    describe('themed', () => {
      it('should match the snapshot', () => {
        const tree = mount(<Spinner />)

        expect(toJson(tree)).toMatchSnapshot()
      })
    })
  })
})

