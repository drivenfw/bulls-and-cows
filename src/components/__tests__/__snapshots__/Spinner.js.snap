// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> snapshot test themed should match the snapshot 1`] = `
.c0 {
  width: 150px;
  height: 150px;
  border-radius: 50%;
  border: 5px solid;
  -webkit-animation: MQNGw 1s linear infinite;
  animation: MQNGw 1s linear infinite;
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;
